// Copyright (c) 2026, Alliance for Open Media. All rights reserved
//
// This source code is subject to the terms of the BSD 3-Clause Clear License
// and the Alliance for Open Media Patent License 1.0. If the BSD 3-Clause Clear
// License was not distributed with this source code in the LICENSE file, you
// can obtain it at www.aomedia.org/license/software-license/bsd-3-c-c. If the
// Alliance for Open Media Patent License 1.0 was not distributed with this
// source code in the PATENTS file, you can obtain it at
// www.aomedia.org/license/patent.

edition = "2023";

package iamf_tools_cli_proto;

option java_multiple_files = true;
option java_outer_classname = "MetadataObu";

message MetadataITUTT35 {
  uint32 itu_t_t35_country_code = 1;
  // Only present if `itu_t_t35_country_code` is 0xFF.
  uint32 itu_t_t35_country_code_extension_byte = 2;
  bytes itu_t_t35_payload_bytes = 3;
}

message Tag {
  string name = 1;
  string value = 2;
}

message MetadataIamfTags {
  repeated Tag tags = 1;
}

message MetadataObuMetadata {
  oneof metadata {
    MetadataITUTT35 metadata_itu_t_t35 = 2;
    MetadataIamfTags metadata_iamf_tags = 3;
  }
}
